tracer/src/Datadog.Trace.Tools.Runner/Checks/ProcessBasicCheck.cs

```diff
@@ -460,5 +497,169 @@ private static bool ParseBooleanConfigurationValue(string value)
+            };
+
+            foreach (var bundleSetupEnding in expectedEndingsForBundleSetup)
+            {
+                if (profilerPathValue.EndsWith(bundleSetupEnding, StringComparison.OrdinalIgnoreCase))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static void NoManageCodeClrVersionIis(ProcessInfo process)
+        {
+            foreach (var module in process.Modules)
+            {
+                var fileName = Path.GetFileName(module);
+
+                if (fileName.Equals("Microsoft.AspNetCore.Server.IIS.dll", StringComparison.OrdinalIgnoreCase))
+                {
+                    AnsiConsole.WriteLine(TracingDotnetOnIis);
+                }
+            }
+        }
+
+        private static string? CheckWindowsInstallation()
+        {
+            string? tracerVersion = null;
+            string? displayName = null;
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                const string datadog64BitProgram = "Datadog .NET Tracer 64-bit";
+                const string datadog32BitProgram = "Datadog .NET Tracer 32-bit";
+
+                var registrySubKeys = new[] { @"SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall", @"SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall" };
+
+                foreach (var registryKey in registrySubKeys)
+                {
+                    using (var key = Registry.LocalMachine.OpenSubKey(registryKey))
+                    {
+                        if (key is not null)
+                        {
+                            foreach (var subKeyName in key.GetSubKeyNames())
+                            {
+                                var subKey = key.OpenSubKey(subKeyName);
+
+                                if (subKey?.GetValue("DisplayName") is datadog64BitProgram or datadog32BitProgram && displayName is null)
+                                {
+                                    displayName = (string?)subKey.GetValue("DisplayName");
+                                    tracerVersion = subKey.GetValue("VersionMajor") + "." + subKey.GetValue("VersionMinor");
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                }
+
+                if (displayName is null)
+                {
+                    Utils.WriteError(TraceProgramNotFound);
+                }
+                else if (displayName is datadog64BitProgram)
+                {
+                    Utils.WriteSuccess(TracerProgramFound(datadog64BitProgram));
+                }
+                else if (displayName is datadog32BitProgram)
+                {
+                    if (RuntimeInformation.OSArchitecture is Architecture.X64)
+                    {
+                        Utils.WriteError(WrongTracerArchitecture(datadog32BitProgram));
```

Comment:
I think this should be a warning if you're checking the OSArchitecture, since it's possible that this app is running as 32-bit. However, if the process bitness is 64-bit and they have only installed the 32-bit MSI, then that should be an error

```suggestion
                        Utils.WriteWarning(WrongTracerArchitecture(datadog32BitProgram));
```